### PR TITLE
hp refinement for bernstein

### PR DESCRIPTION
**Pull request by mtezzele**
hp implementation for FE_Bernstein with test analogous to the one for FE_Q.

### CODE REVIEW

File: include/deal.II/fe/fe_bernstein.h [hunk 1]
```diff
@@ -72,6 +72,88 @@ class FE_Bernstein : public FE_Q_Base<TensorProductPolynomials<dim>,dim,spacedim
   FE_Bernstein (const unsigned int p);
 
   /**
+  * Return the matrix interpolating from a face of of one element to the face
```
Comment (luca-heltai) [inline include/deal.II/fe/fe_bernstein.h]: remove on "of"

File: include/deal.II/fe/fe_bernstein.h [hunk 2]
```diff
@@ -72,6 +72,88 @@ class FE_Bernstein : public FE_Q_Base<TensorProductPolynomials<dim>,dim,spacedim
...
+  * the same type and FE_Nothing. For all other elements, an exception of
+  * type FiniteElement<dim,spacedim>::ExcInterpolationNotImplemented is
+  * thrown.
+  */
+  virtual void
+  get_face_interpolation_matrix (const FiniteElement<dim,spacedim> &source,
+                                 FullMatrix<double>       &matrix) const;
+
+  /**
+   * Return the matrix interpolating from a face of of one element to the face
```
Comment (luca-heltai) [inline include/deal.II/fe/fe_bernstein.h]: Same here.

**Comment by luca-heltai** [comment]
I'm not an expert on the HP infrastructure. Maybe someone else should take a look at this...

File: source/fe/fe_bernstein.cc [hunk 3]
```diff
@@ -41,6 +44,201 @@ FE_Bernstein<dim,spacedim>::FE_Bernstein (const unsigned int degree)
...
+    }
+}
+
+
+template <int dim, int spacedim>
+std::vector<std::pair<unsigned int, unsigned int> >
+FE_Bernstein<dim,spacedim>::hp_line_dof_identities (const FiniteElement<dim,spacedim> &fe_other) const
+{
+  // There are not any support points on line.
+  return std::vector<std::pair<unsigned int, unsigned int> > ();
```
Comment (bangerth) [inline source/fe/fe_bernstein.cc]: I think I don't understand this (and the same for the quads below). If there are no dofs located on lines, how can you have higher order elements?

**Comment by bangerth** [comment]
Looks mostly good, but I can't figure out why there are no dofs located on lines for this element. This may actually be correct, but I don't understand enough about the element to know -- though, if it is correct, maybe you can extend the comment :-)

**Comment by mtezzele** [comment]
Since this fe is not interpolatory but on the vertices, we can not identify dofs on lines and on quads even if there are dofs on lines and on quads.
I proceed adding comments.

**Comment by bangerth** [comment]
OK, this is clearer now -- thanks for updating the comment!

**Comment by bangerth** [comment]
Can you please squash these commits into one?

**Comment by mtezzele** [comment]
Done.

**Pull request merged**